ext/nnstreamer/tensor_decoder/tensordec-flatbuf.cc

```diff
@@ -0,0 +1,159 @@
+
+namespace nnstreamer {
+namespace flatbuf {
+
+void init_fb (void) __attribute__ ((constructor));
+void fini_fb (void) __attribute__ ((destructor));
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+fb_init (void **pdata)
+{
+  *pdata = NULL;
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static void
+fb_exit (void **pdata)
+{
+  return;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+fb_setOption (void **pdata, int opNum, const char *param)
+{
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstCaps *
+fb_getOutCaps (void **pdata, const GstTensorsConfig * config)
+{
+  return gst_caps_from_string (GST_FLATBUF_TENSOR_CAP_DEFAULT);
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstFlowReturn
+fb_decode (void **pdata, const GstTensorsConfig * config,
+    const GstTensorMemory * input, GstBuffer * outbuf)
+{
+  char *name;
+  Tensor_type type;
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  unsigned int i, num_tensors = config->info.num_tensors;
+  flatbuffers::uoffset_t fb_size;
+  flatbuffers::FlatBufferBuilder builder;
+  std::vector<flatbuffers::Offset<Tensor>> tensor_vector;
+  flatbuffers::Offset<flatbuffers::Vector<uint32_t>> dim;
+  flatbuffers::Offset<flatbuffers::String> tensor_name;
+  flatbuffers::Offset<flatbuffers::Vector<unsigned char>> input_vector;
+  flatbuffers::Offset<Tensor> tensor;
+  flatbuffers::Offset<Tensors> tensors;
+  frame_rate fr = frame_rate (config->rate_n, config->rate_d);
+
+  /* Fill the info in tensor and puth to tensor vector */
+  for (i = 0; i < num_tensors; i++) {
+    unsigned char *tmp_buf;
+
+    dim = builder.CreateVector (config->info.info[i].dimension, NNS_TENSOR_RANK_LIMIT);
+    name = config->info.info[i].name;
+  
+    if (name == NULL)
+      tensor_name = builder.CreateString ("Anonymous");
+    else
+      tensor_name = builder.CreateString (name);
+
+    type = (Tensor_type) config->info.info[i].type;
+
+    /* Create the vector first, and fill in data later */
+    /** @todo Consider to remove memcpy */
+    input_vector = builder.CreateUninitializedVector<unsigned char> (input[i].size, &tmp_buf);
+    memcpy (tmp_buf, input[i].data, input[i].size);
+
+    tensor = CreateTensor (builder, tensor_name, type, dim, input_vector);
+    tensor_vector.push_back (tensor);
+  }
+
+  tensors = CreateTensors (builder, num_tensors, &fr, builder.CreateVector (tensor_vector));
+
+  /* Serialize the data.*/
+  builder.Finish (tensors);
+  fb_size = builder.GetSize ();
+
+  g_assert (outbuf);
+
+  if (gst_buffer_get_size (outbuf) == 0)
+    out_mem = gst_allocator_alloc (NULL, fb_size, NULL);
+  else {
+    if (gst_buffer_get_size (outbuf) < fb_size) {
+      gst_buffer_set_size (outbuf, fb_size);
+    }
+    out_mem = gst_buffer_get_all_memory (outbuf);
+  }
+  g_assert (gst_memory_map (out_mem, &out_info, GST_MAP_WRITE));
```

Comment:
How about adding error handling codes, not `g_assert`

Reply:
Thank you. I added an error handling code.